sdk/python/kfp_tekton/compiler/_data_passing_rewriter.py

```diff
@@ -764,3 +764,25 @@ def _append_original_pr_name_env(task_template):
     for step in task_template['taskSpec']['steps']:
         if step['name'] == 'main':
             _append_original_pr_name_env_to_step(step)
+
+
+def fix_big_data_passing_using_volume(workflow, pipeline_conf):
+    if workflow['spec'].get('workspaces'):
+        if pipeline_conf.data_passing_method._volume != None:
+            volume_dict = pipeline_conf.data_passing_method._volume.to_dict()
+            if volume_dict.get('persistent_volume_claim'):
+                for workspace in workflow['spec']['workspaces']:
+                    if workspace['name'] == workflow['metadata']['name']:
+                        workspace.pop('volumeClaimTemplate')
+                        temp_vc = {}
+                        for key, value in volume_dict.get('persistent_volume_claim').items():
+                            if value != None:
+                                if key == 'claim_name':
+                                    temp_vc['claimName'] = value
+                                if key == 'read_only':
+                                    temp_vc['readOnly'] = value
+                            workspace['persistentVolumeClaim'] = temp_vc
```

Comment:
Mapping lookup + if seems more clear here.

`temp_vc` seems unneeded, as considering the loop only iterates once, it looks like it's never accumulating anything but just holding the single information above (either `claimName` or `readOnly`), so it could even be inlined as a dict literal.

Reply:
In the test case there's only one key, but v1volume can have both claim_name and read_only (so can iterate up to two times and can be more with future changes in v1volume class).